lib/api.js

```diff
@@ -113,11 +146,38 @@ function makeApiRequest(transport, opts, bodyObj, callback) {
       respData = respData.join('');
       parseApiResponse(respData, callback);
     });
+
+    dequeueApiRequest(req);
   });
 
+  pendingQueue.push(req);
+
+  var originalArguments = arguments;
   req.on('error', function (err) {
-    logger.error('Could not make request to rollbar, ' + err);
-    callback(err);
+    dequeueApiRequest(req);
```

Comment:
This is called when the request errors; and success and error are the only two possibilities, so exactly one of these two callbacks will be called and thus the api request will always be dequeued from the pending queue.